File: set1/3/Main.java
import java.util.*;
import java.lang.*;
import java.io.*;

public class Main {
    public static void main(String[] args) {
        FastReader rd = new FastReader();
        
        int N = rd.nextInt();
        int []arr = new int[N];
        for(int i = 0; i < N; ++i) arr[i] = rd.nextInt();
        int X = rd.nextInt();

        long sum = 0, cnt = 0;
        for(int i = 0; i < N; ++i) {
            if(gcd(arr[i], X) == 1) {
                sum += arr[i];
                cnt ++;
            }
        }

        System.out.printf("%.6f", (double)sum / cnt);
    }

    static long gcd(long a, long b) {
        if(b == 0) return a;
        return gcd(b, a % b);
    }

    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while(st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                }
                catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() { return Integer.parseInt(next()); }
        long nextLong() { return Long.parseLong(next()); }
        double nextDouble() { return Double.parseDouble(next()); }
        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            }
            catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}

File: set1/3/main.cpp
#include<iostream>
#include<vector>
#include<iomanip>

using namespace std;

long long gcd(long long a, long long b) {
    if(b == 0) return a;
    return gcd(b, a % b);
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int N; cin >> N;
    vector<int> arr(N);
    for(int i = 0; i < N; ++i) cin >> arr[i];
    int X; cin >> X;

    long long sum = 0, cnt = 0;
    for(int i = 0; i < N; ++i) {
        if(gcd(arr[i], X) == 1) {
            sum += arr[i];
            ++ cnt;
        }
    }

    cout << setprecision(6) << fixed << (double)sum / cnt;

    return 0;
}